awscli/clidriver.py

```diff
@@ -484,7 +484,7 @@ def __call__(self, args, parsed_globals):
         event = 'before-building-argument-table-parser.%s.%s' % \
             (self._parent_name, self._name)
         self._emit(event, argument_table=self.arg_table, args=args,
-                   session=self._session)
+                   session=self._session, parsed_globals=parsed_globals)
```

Comment:
after digging into the code, it made more sense to add the handler during the `before-building-argument-table-parser` event over `building-command-table` because of the availability of `parsed_globals` in the function call. It could be done but would require more changes to the code by passing around the variable to a couple of different functions or setting it as an instance variable.